pkg/apis/numaflow/v1alpha1/edge_types.go

```diff
@@ -39,7 +39,17 @@ type Edge struct {
 }
 
 type ForwardConditions struct {
-	KeyIn []string `json:"keyIn" protobuf:"bytes,1,rep,name=keyIn"`
+	// Tags used to specify tags for conditional forwarding
+	Tags *TagConditions `json:"tags" protobuf:"bytes,1,rep,name=tags"`
+}
+
+type TagConditions struct {
+	// Operator specifies the type of operation that should be used for conditional forwarding
+	// value could be "and", "or", "not"
+	// +optional
+	Operator string `json:"operator" protobuf:"bytes,1,opt,name=operator"`
```

Comment:
Define a type e.g. `type LogicOperator string`, and some constants.

Use `Operator *LogicOperator xxxx`

Add a function and use it in other places.

```golang
func (tc TagConditions) GetOperator() LogicOperator {

}
```